onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -803,6 +809,40 @@ def _infer_Conv(self, node):
             helper.make_tensor_value_info(node.output[0], vi.type.tensor_type.elem_type,
                                           get_shape_from_sympy_shape(sympy_shape)))
 
+    def _infer_Einsum(self, node):
+        shape_0 = self._get_sympy_shape(node, 0)
+        shape_1 = self._get_sympy_shape(node, 1)
+        shape_0_len = len(shape_0)
+        shape_1_len = len(shape_1)
+
+        equation = get_attribute(node, 'equation')  # e.g b'abc,cde->abde'
+
+        assert (equation[shape_0_len] == 44)  # ','
```

Comment:
The assertion is not always True.

I saw some example in https://github.com/onnx/onnx/blob/master/docs/Operators.md#Einsum like:
```
'...ii ->0...i'
'ij->i'0
'ij->ji'
'i,i'
```
I am not sure whether onnx library supports symbolic shape inference for Einsum. If so, it will be better to use that.

Reply:
https://github.com/onnx/onnx/blob/623dfaa0151b2e4ce49779c3ec31cbd78c592b80/onnx/defs/math/defs.cc#L3275 I saw something here: not be able to directly apply but could take as a reference